src/plugins/discover/public/application/components/json_code_editor/json_code_editor.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback } from 'react';
+import { i18n } from '@kbn/i18n';
+import { monaco, XJsonLang } from '@kbn/monaco';
+import { EuiButtonEmpty, EuiCopy, EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
+import { CodeEditor } from '../../../../../kibana_react/public';
+
+const codeEditorAriaLabel = i18n.translate('discover.json.codeEditorAriaLabel', {
+  defaultMessage: 'Read only JSON view of an elasticsearch document',
+});
+const copyToClipboardLabel = i18n.translate('discover.json.copyToClipboardLabel', {
+  defaultMessage: 'Copy to clipboard',
+});
+
+export const JsonCodeEditor = (value: any) => {
```

Comment:
Would be nice to omit `any` here, because what you do here is, instead of `props` you're using value. then you're using the component <JsonCodeEditor value={valueFromOutside} />. So `value.valueFromOutside`  would be needed to get this value, which could be easily omitted by e.g.
```
interface JsonCodeEditorProps {
 value: ElasticSearchHit;
}
```